ci/kokoro/macos/upload-cache.sh

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+if [[ $# != 2 ]]; then
+  echo "Usage: $(basename "$0") <cache-folder> <cache-name>"
+  exit 1
+fi
+
+readonly CACHE_FOLDER="$1"
+readonly CACHE_NAME="$2"
+
+KOKORO_GFILE_DIR="${KOKORO_GFILE_DIR:-/private/var/tmp}"
+readonly KOKORO_GFILE_DIR
+
+readonly KEYFILE="${KOKORO_GFILE_DIR}/build-results-service-account.json"
+if [[ ! -f "${KEYFILE}" ]]; then
+  echo "================================================================"
+  echo "Service account for cache access is not configured."
+  echo "No attempt will be made to download the cache, exit with success."
+  exit 0
+fi
+
+if [[ "${KOKORO_JOB_TYPE:-}" == "PRESUBMIT_GERRIT_ON_BORG" ]] || \
+   [[ "${KOKORO_JOB_TYPE:-}" == "PRESUBMIT_GITHUB" ]]; then
+  echo "================================================================"
+  echo "This is a presubmit build, cache will not be updated, exist with success."
+# TODO(coryan) - warm up the cache on first PR  exit 0
```

Comment:
Is that exit supposed to be commented out like it it's?

Reply:
No. The first time I send a PR for a cached build I need to warm up the cache *event on PRs*, then I remove this comment before merging so future PRs do not dirty up the cache.

Reply:
Woops, I missed that this was a draft earlier.